lib/ain-evm/src/core.rs

```diff
@@ -816,6 +816,91 @@ impl EVMCoreService {
 
         Ok((listener.trace, execution_success, return_value, used_gas))
     }
+
+    pub fn create_access_list(&self, arguments: EthCallArgs) -> Result<(AccessList, u64)> {
+        let EthCallArgs {
+            caller,
+            to,
+            value,
+            data,
+            gas_limit,
+            access_list,
+            block_number,
+            ..
+        } = arguments;
+
+        let block_header = self
+            .storage
+            .get_block_by_number(&block_number)?
+            .ok_or_else(|| format_err!("Block not found"))
+            .map(|block| block.header)?;
+        let state_root = block_header.state_root;
+        debug!(
+            "Calling EVM at block number : {:#x}, state_root : {:#x}",
+            block_number, state_root
+        );
+
+        let mut vicinity = Vicinity::from(block_header);
+        vicinity.origin = caller;
+        vicinity.gas_price = arguments.gas_price;
+
+        let backend = EVMBackend::from_root(
+            state_root,
+            Arc::clone(&self.trie_store),
+            Arc::clone(&self.storage),
+            vicinity,
+            None,
+        )
+        .map_err(|e| format_err!("Could not restore backend {}", e))?;
+
+        static CONFIG: Config = Config::shanghai();
+        let metadata = StackSubstateMetadata::new(gas_limit, &CONFIG);
+        let state = MemoryStackState::new(metadata.clone(), &backend);
+        let al_state = MemoryStackState::new(metadata, &backend);
+        let precompiles = MetachainPrecompiles;
+        let mut executor = StackExecutor::new_with_precompiles(state, &CONFIG, &precompiles);
+        let mut al_executor = StackExecutor::new_with_precompiles(al_state, &CONFIG, &precompiles);
+
+        let mut listener = crate::eventlistener::StorageAccessListener::new();
+
+        runtime_using(&mut listener, move || {
```

Comment:
Does the RPC depend on the execution result of the transaction call?

Reply:
https://github.com/DeFiCh/ain/blob/117b9cd26aa25a4b8813f4b967670eb759cb2b28/lib/ain-evm/src/core.rs#L888-L895

Reply:
This doesnt resolve anything. The result of the transaction_call from the executor is never processed.

Reply:
We are making some assumptions that the execution result is successful which this pipeline isnt covered